hnn_core/cell.py

```diff
@@ -62,19 +62,22 @@ class _Cell(ABC):
     soma_props : dict
         The properties of the soma. Must contain
         keys 'L', 'diam', and 'pos'
+    record_vsoma : bool
+        Option to record somatic voltages from cells
 
     Attributes
     ----------
     pos : list of length 3
         The position of the cell.
     """
 
-    def __init__(self, gid, soma_props):
+    def __init__(self, gid, soma_props, record_vsoma=0):
```

Comment:
I think what you need to do is have `record_vsoma` as an argument to `simulate_dipole` and then pass it down the functions until you instantiate it. Then you don't need this parameter to the `Cell` class. You can do something like:

```py
cell = L5Pyr()
if record_vsoma: 
    cell.record_voltage_soma()
```

It's a bit odd to have it for individual cells since we're going to enable or disable it for all cells. I presume things will break if you set `record_vsoma=True` for some cells and `False` for others.

Reply:
I don't believe it would break, the `vsoma` would just return an empty list for gids with `record_vsoma=False`. 

I agree with passing this as an argument to `simulate_dipole`, but it could be beneficial to have an option to select specific cells to record from in the interest of simulation time. I'll go with your recommendation for now to get this PR moving, and consider the pros/cons of enabling recordings for all cells in the current recording PR.